Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift

```diff
@@ -0,0 +1,486 @@
+                // http方法
+                switch method {
+                case .post:
+                    request.httpMethod = "POST"
+                case .get:
+                    request.httpMethod = "GET"
+                case .put:
+                    request.httpMethod = "PUT"
+                }
+                // 开始请求
+                URLSession.shared.dataTask(
+                    with: request
+                ) { data, response, error in
+                    if let statusCode = (response as? HTTPURLResponse)?
+                        .statusCode, statusCode != 200 {
+                        completion(.failure(.errorWithCode(statusCode)))
+                    }
+                    print(error ?? "ErrorInfo nil")
+                    if let error = error {
+                        completion(.failure(.dataTaskError(
+                            error
+                                .localizedDescription
+                        )))
+                        print(
+                            "DataTask error in General HttpMethod: " +
+                                error.localizedDescription + "\n"
+                        )
+                    } else {
+                        guard let data = data else {
+                            completion(.failure(.noResponseData))
+                            print("found response data nil")
+                            return
+                        }
+                        guard response is HTTPURLResponse else {
+                            completion(.failure(.responseError))
+                            print("response error")
+                            return
+                        }
+                        DispatchQueue.main.async {
+                            let decoder = JSONDecoder()
+                            decoder.keyDecodingStrategy = keyDecodingStrategy
+
+//                            let dictionary = try? JSONSerialization.jsonObject(with: data)
+//                            print(dictionary ?? "None")
+
+                            do {
+                                let requestResult = try decoder.decode(
+                                    T.self,
+                                    from: data
+                                )
+                                completion(.success(requestResult))
+                            } catch {
+                                print(error)
+                                completion(.failure(.decodeError(
+                                    error
+                                        .localizedDescription
+                                )))
+                            }
+                        }
+                    }
+                }.resume()
+            }
+        }
+    }
+
+    /// 返回需要POST的结果接口
+    /// - Parameters:
+    ///   - method:Method, http方法的类型
+    ///   - url:String，请求的路径
+    ///   - completion:异步返回处理好的data以及报错的类型
+    ///
+    ///  需要自己传URL类型的url过来
+    static func postRequest(
```

Comment:
refactor this whole func.
- use `async`
- no need to set header like `x-rpc-client_type` etc
- no need to set `method` here
- delete those `print`
-